lib/ansible/plugins/inventory/scaleway.py

```diff
@@ -156,7 +157,10 @@ class InventoryModule(BaseInventoryPlugin):
     NAME = 'scaleway'
 
     def verify_file(self, path):
-        return "scaleway" in path
+        with open(path) as f:
+            raw_file = f.read()
+            d = from_yaml(raw_file)
+            return d.get("plugin") == "scaleway"
```

Comment:
[`BaseInventoryFilePlugin._read_config_data`](https://github.com/ansible/ansible/blob/73665f379f657c4844ca275f1940bcb28053b148/lib/ansible/plugins/inventory/__init__.py#L186) perform the same things (reading the file, checking the `plugin` key), I don't see any reason to no reusing it. I propose something like:
```
def verify_file(self, path):
    if super(BaseInventoryFilePlugin, self).verify_file(path):
        try:
            self._read_config_data(path)
            return True
        except Exception:
            pass

    return False
```